libs/labelbox/src/labelbox/schema/member.py

```diff
@@ -0,0 +1,433 @@
+from typing import Optional
+from labelbox.exceptions import (
+    MalformedQueryException,
+    ResourceNotFoundError,
+    UnprocessableEntityError,
+)
+from typing import Set, Iterator, Any, List
+from pydantic import (
+    Field,
+    field_validator,
+    model_serializer,
+    model_validator,
+)
+from labelbox.utils import _CamelCaseMixin
+from labelbox.schema.role import Role
+from labelbox import Client
+
+
+class ProjectMembership(_CamelCaseMixin):
+    """Represents a members project role
+
+    Args:
+        project_id (str): id of the project you want the member included
+        role (Optional[Role]): Members role for the project. None represents the member having a default role.
+    """
+
+    project_id: str
+    role: Optional[Role] = Field(default=None)
+
+    def __hash__(self) -> int:
+        return self.project_id.__hash__()
+
+    @model_serializer()
+    def serialize_model(self):
+        return {
+            "projectId": self.project_id,
+            "roleId": None if self.role is None else self.role.uid,
+        }
+
+
+class Member(_CamelCaseMixin):
```

Comment:
SDK already have [class User](https://github.com/Labelbox/labelbox-python/blob/a75c402c2a6854b59ebab61d4950db5cb1202db7/libs/labelbox/src/labelbox/schema/user.py#L9). What would be the difference? Can be `User` class just refactored to have more fields and support for new methods? Otherwise there is again this weird state where SDK going to have two types to represent the same object.

Reply:
I attempted this it does not work since we have to use a GetMemberRework call to make sure we get all the fields including ProjectMemberships. We also need to setUserAccess Call which is not supported by the built in entities user groups had a similar issues

Reply:
Also this is 100% experimentally but right now the SDK is severally lacking on member management so for Alignerr for example i have to heavily use GQL direclty